astropy/io/ascii/core.py

```diff
@@ -1042,7 +1042,8 @@ def _convert_vals(self, cols):
                                   else self.default_converters)
 
             # If the user supplied a specific convert then that takes precedence over defaults
-            converters = self.converters.get(col.name, default_converters)
+            converters = self.converters[col.name] \
```

Comment:
My implementation was obviously one of the really stupid ideas that just looked too simple because they are. On the other hand, what @mhvk suggests here is also not quite right, I think. At this point, there is `col.dtype` is always `None` (unless we also allow passing in a dtype as a new argument -- but that's an API change I'd like to leave for #11278). In fact, the whole problem is that `if col.name in self.converters` will be `False` until we try to call `self.converters[col.name]` at least once.

Reply:
I think that @hamogu's implementation (which is close to the original in using a separate `default_converters` ) is logically easier to follow and easier to read.